src/lua_console.cpp

```diff
@@ -0,0 +1,86 @@
+#include "lua_console.h"
+
+#include "catalua.h"
+#include "catacharset.h"
+
+lua_console::lua_console() : cWin( newwin( lines, width, 0, 0 ) ), iWin( newwin( 1, width,lines, 0 ) )
+{
+}
+
+void lua_console::dispose()
+{
+    werase( cWin );
+    werase( iWin );
+    delwin( cWin );
+    delwin( iWin );
+}
+
+std::string lua_console::get_input()
+{
+    long key = 0;
+    int pos = -1;
+    return string_input_win( iWin, "", width, 0, 0, width, true, key, pos, "LUA", 0, lines, true, false, callbacks );
+}
+
+void lua_console::draw()
+{
+    werase( cWin );
+
+    // Some juggling to make sure text is aligned with the bottom of the console.
+    int stack_size = text_stack.size() - scroll;
+    for( int i = lines; i > lines - stack_size && i >= 0; i-- ) {
+        auto line = text_stack[stack_size - 1 - ( lines - i )];
+        mvwprintz( cWin, i - 1, 0, line.second, line.first.c_str() );
```

Comment:
This should be `"%s", line.first.c_str()`. I can currently crash the game easily by entering `print("%s %s %s")`.